### PR TITLE
Created crud controller for categories, model and association one to …

### LINKED ISSUE #9: Create CRUD for adding Categories and connect it with Products

### PR DESCRIPTION
**Pull request by matsevich**
Created crud controller for categories, model. Association one to many  with products
[Create CRUD for adding Categories and connect it with Products](https://github.com/matsevich/trucco/projects/1#card-52764701)
Screenshot from 2021-02-13 17-06-16
Screenshot from 2021-02-13 16-49-10
Screenshot from 2021-02-13 16-49-35
Screenshot from 2021-02-13 16-58-52
Screenshot from 2021-02-13 16-59-13
Screenshot from 2021-02-13 16-59-19

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:success] = "You have successfully created new '#{@category.title}' category"
```
Comment (denys281) [inline app/controllers/categories_controller.rb]: I think you should use I18n from the beginning

File: app/views/categories/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+.d-flex.justify-content-center
+  h2.mr-4 All Categories
+  = link_to new_category_path
+    button.btn.btn-outline-secondary
+      = fa_icon 'plus'
+table.table
+  thead.thead
+    tr
+      th
+        span Title
```
Comment (denys281) [inline app/views/categories/index.html.slim]: I18n

File: app/controllers/categories_controller.rb (hunk 1 above)
Comment (matsevich) [reply app/controllers/categories_controller.rb]: @denys281  ok, I was just thinking about it

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  it { is_expected.to validate_presence_of(:title) }
```
Comment (denys281) [inline spec/models/category_spec.rb:4]: It is not necessary.

File: config/initializers/locale.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+I18n.available_locales = %i[en ua]
```
Comment (denys281) [inline config/initializers/locale.rb]: I think uk)

File: spec/models/category_spec.rb (hunk 3 above)
Comment (matsevich) [reply spec/models/category_spec.rb:4]: @denys281  I don't want the ability to create categories without titles, so I prefer to leave that validation, and also this test

File: spec/controllers/categories_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe CategoriesController, type: :controller do
```
Comment (denys281) [inline spec/controllers/categories_controller_spec.rb:3]: Rails recommends to use requests specs ) https://medium.com/just-tech/rspec-controller-or-request-specs-d93ef563ef11

File: spec/models/category_spec.rb (hunk 3 above)
Comment (denys281) [reply spec/models/category_spec.rb:4]: As you wish) but I prefer not to write specs for relations and validations)

**Review by denys281 — APPROVED** [review]

File: spec/controllers/categories_controller_spec.rb (hunk 5 above)
Comment (matsevich) [reply spec/controllers/categories_controller_spec.rb:3]: @denys281 ok, I'll use requests specs for my future controllers

**Pull request merged**